saml/src/test/java/com/linecorp/armeria/server/saml/SamlAssertionConsumerFunctionTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.saml;
+
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.opensaml.messaging.context.MessageContext;
+import org.opensaml.saml.saml2.core.Response;
+
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class SamlAssertionConsumerFunctionTest {
+
+    @Test
+    void testServeWithNullMessageContext() {
+        // given
+        ServiceRequestContext ctx = mock(ServiceRequestContext.class);
+        AggregatedHttpRequest req = mock(AggregatedHttpRequest.class);
```

Comment:
You don't need to mock these two types. Please use:

- `ServiceRequestContext.of()` or `.builder()`; and
- `AggregatedHttpRequest.of()`.